### PR TITLE
Create invoice on sponsor acceptance

### PR DESCRIPTION
**Pull request by karianne**

### CODE REVIEW

File: app/decorators/sponsor_accepted_slack_notifier.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  private
+  def notify_slack
+    if status_was_changed_to_accepted
+      count = Sponsor.where(status: 'accepted').count
+      body = {
+          :channel => '#sponsors',
+          :text => "*Good news everyone!* #{@sponsor.name} has agreed to become a partner! We now have #{count} partners."
+      }
+      SlackNotifier.post_to_slack(body)
```
Comment (oma) [inline app/decorators/sponsor_accepted_slack_notifier.rb]: liker :+1:

File: app/decorators/sponsor_ticket_creator.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+class SponsorTicketCreator < SimpleDelegator
+
+  def initialize(sponsor)
+    @old_status = sponsor.status
+    @sponsor = sponsor
+    super(@sponsor)
```
Comment (oma) [inline app/decorators/sponsor_ticket_creator.rb]: Jeg liker disse klassene! 

Nerd alert: Jeg tror du kan skrive kun `super` så vil den send inn samme som init, nemlig `sponsor` (ikke instansvariabelen)

File: app/controllers/sponsors_controller.rb [hunk 3]
```diff
@@ -37,6 +36,14 @@ def create
   end
 
   def update
+    @sponsor = SponsorAcceptedSlackNotifier.new(
```
Comment (oma) [inline app/controllers/sponsors_controller.rb]: you can take this even a few steps further. You know what I'm going to say, this controller method knows too much. It should pass current_user, params :id and :sponsor to a factory/create method that returns the object you want in the state you want :)

File: app/decorators/sponsor_ticket_creator.rb (hunk 2 above)
Comment (karianne) [inline app/decorators/sponsor_ticket_creator.rb]: Det har du helt rett i!

File: app/controllers/sponsors_controller.rb (hunk 3 above)
Comment (karianne) [inline app/controllers/sponsors_controller.rb]: I see what you mean - and I tried doing it - but I decided against it for readability reasons. I want it to be very clear what actually happens when you update a sponsor, while the implementation is hidden away other places.

**Pull request merged**